### PR TITLE
add jungles/nojungles flags for mapgen v7

### PR DESCRIPTION
**Pull request by zeuner**
Since I saw insanely slow rendering speeds for jungle biomes on slow machines, I found the possibility to disable jungles in mgv6 rather valuable.

This fork adds the possibility to disable jungles for mgv7.

Some thought: Is there a reason why there is no minetest.unregister_biome?

### CODE REVIEW

File: mods/default/mapgen.lua [hunk 1]
```diff
@@ -494,6 +494,8 @@ end
 
 -- All mapgens except mgv6
 
+local nojungles = nil
```
Comment (sfan5) [inline mods/default/mapgen.lua]: `= false`

File: mods/default/mapgen.lua [hunk 2]
```diff
@@ -1058,65 +1060,69 @@ function default.register_biomes(upper_limit)
...
-		--node_water = "",
-		--node_river_water = "",
-		node_riverbed = "default:sand",
-		depth_riverbed = 2,
-		y_min = 1,
-		y_max = upper_limit,
-		heat_point = 86,
-		humidity_point = 65,
-	})
+	if nil == nojungles then
```
Comment (sfan5) [inline mods/default/mapgen.lua]: `not nojungles`

**Comment by paramat** [comment]
:-1:
Rainforest is no worse than pine trees, which are twice as common and worse for rendering due to snow nodeboxes on the leaves. If you have rendering problems us 'leaves_style = opaque' or reduce view range.

Removing forests isn't the right approach, but if you want to do that you can write a mod (i'll write it for you if you want) that re-registers the biomes without rainforest and/or pine trees. The official intended way to alter biomes is to use a mod in this way.

Removing an integral part of the biome system is a crude way to approach this that may have been ok in a mess like mgv6 but not in other mapgens. It had effect in mgv6 because the only other trees in mgv6 were low density small appletrees.

**Comment by paramat** [comment]
This being applied to mgv7 only is incorrect, it needs to apply to all mapgens that use the biome API, all mapgens except mgv6. But don't bother as i strongly oppose this. Hmmmm also does not like making the engine side of mapgen unnecessarily complex and believes that re-registering biomes in a mod should be used in cases like this.

**Comment by zeuner** [comment]
@paramat: In my case, jungles did not fit too well thematically, anyway, so removing them does seem like a reasonable approach. I agree with most of your critique, though. Removing unwanted biomes from a mod is more flexible, and keeps the configuration simple.

I do think there are good reasons to only remove (and maybe replace) single biomes, so I don't think the mod author should have to bother with biome list implementation details, or re-registering all the biomes not touched. So, I would now propose adding a suitable function to the API (https://github.com/minetest/minetest/pull/5445).

@sfan5: Just in case someone likes my initial approach, I adopted your stylistic suggestions.

**Comment by paramat** [comment]
The new request seems fairly reasonable. When a biome is removed it is best to move the others around, but this is complex work that would be beyond many players.

**Comment by paramat** [comment]
Yes it is reasonable to remove single biomes, my earlier comments were more directed to removing a biome because of rendering problems.

**Comment by zeuner** [comment]
I agree that there are more straight-forward ways to handle rendering problems. But if the more computationally expensive biomes are not relevant to a given theme anyway, why not allow the user to experience fancier leaves rendering etc. if the issue can be handled by removing the less relevant biomes?

Anyway, if there are no objections, I would close this PR in favour of the other request.

**Comment by paramat** [comment]
Yes i agree, it's reasonable.

**Pull request closed without merging**